lightning/src/ln/channelmanager.rs

```diff
@@ -3478,6 +3535,24 @@ impl<ChanSigner: ChannelKeys, M: Deref + Sync + Send, T: Deref + Sync + Send, K:
 	}
 }
 
+/// `Persist` defines behavior for persisting the ChannelManager: this could mean writing once to
+/// disk, and/or uploading to one or more backup services. This persistence should probably happen
+/// in a background thread for running an efficient node, rather than in real time as updates occur.
+/// See [`block_until_needs_persist`] for more details.
+/// Note that restoring an outdated ChannelManager from disk runs the risk of force-closing
+/// channels.
+///
+///[`ChannelManager::block_until_needs_persist`]: struct.ChannelManager.html#method.block_until_needs_persist
+pub trait Persist<ChanSigner: 'static + ChannelKeys + Writeable, M: 'static + chain::Watch<Keys=ChanSigner>, T: 'static + BroadcasterInterface, K: 'static + KeysInterface<ChanKeySigner=ChanSigner>, F: 'static + FeeEstimator, L: 'static + Logger>: Send + Sync {
```

Comment:
Does this not belong in the persister crate? Its never used in the `lightning` crate and it doesn't need any access to lightning-specific stuff. Trying to use it without the persister crate also doesn't seem like it would be useful - its really a way to describe what `BackgroundProcessor` will call.

Reply:
Could a similar argument be made about `BackgroundProcessor` given that isn't solely used for persisting any longer?

Reply:
It does seem like `BackgroundProcessor` might belong in its own crate? Then it could live with this ChannelManager persistence trait (and in the future, some channel graph persistence trait).

Then `lightning-persister` could maybe become `monitor-persister`, or `critical-persister` since that's all it's doing?

Reply:
I was thinking more towards moving `BackgroundProcessor` into the `lightning` crate such that all the `Persist` traits can uniformly live next to what they persist. Then the `lightning-persister` crate would contain `FilesystemPersister` which would implement those traits.

Reply:
I do still wonder whether it makes sense to just replace it with a typedef for a function instead.

Reply:
So, function aliases seem to not work because the function needs to have an `&self` parameter, to get to the data storage path. I think I'm going to just move the trait to the `BackgroundProcessor` crate.

Reply:
Update: switched to a function alias.